src/Thruway/Role/Subscriber.php

```diff
@@ -219,13 +251,49 @@ public function subscribe(ClientSession $session, $topicName, callable $callback
             "options"    => $options,
             "deferred"   => $deferred
         ];
-
+        
         array_push($this->subscriptions, $subscription);
-
+        
         $subscribeMsg = new SubscribeMessage($requestId, $options, $topicName);
         $session->sendMessage($subscribeMsg);
 
         return $deferred->promise();
     }
-
+	
+    /**
+     * process unsubscribe
+     * @param ClientSession $session
+     * @param string $subscriptionId
+     * @return boolean
+     */
+    public function unsubscribe(ClientSession $session, $subscriptionId)
+    {
+        $requestId = Utils::getUniqueId();
+        $subscriptionExists = false;
+        $deferred  = new Deferred();
+        
+        foreach ($this->subscriptions as $i => $subscription) {
+            if ($subscription["subscription_id"] == $subscriptionId) {
+                $subscriptionExists = true;
+                
```

Comment:
We should noop the subscription callback at this point so that the callback won't receive messages once it has asked to be unsubscribed - even if the router decides to keep sending them.